src/http/status.rs

```diff
@@ -406,113 +290,350 @@ async fn handle_metrics(
+    );
+    res.single("http_bytes_read", server_metrics.bytes_read());
 
-    // http_bytes_written
-    writeln!(res,
-        "\n\
-        # HELP routinator_http_bytes_written number of bytes written via HTTP\n\
-        # TYPE routinator_http_bytes_written counter"
-    ).unwrap();
-    writeln!(res,
-        "routinator_http_bytes_written {}", server_metrics.bytes_written()
-    ).unwrap();
+    res.header(
+        "http_bytes_written", "counter",
+        "number of bytes written to HTTP connections"
+    );
+    res.single("http_bytes_written", server_metrics.bytes_written());
 
-    // http_requests
-    writeln!(res,
-        "\n\
-        # HELP routinator_http_requests number of bytes written via HTTP\n\
-        # TYPE routinator_http_requests counter"
-    ).unwrap();
-    writeln!(res,
-        "routinator_http_requests {}", server_metrics.requests()
-    ).unwrap();
+    res.header(
+        "http_requests", "counter",
+        "number of received HTTP requests"
+    );
+    res.single("http_requests", server_metrics.requests());
+
+    res.into_response()
+}
 
 
-    Response::builder()
+#[derive(Default)]
+struct PrometheusMetrics {
+    target: String,
+    prefix: &'static str,
+}
+
+impl PrometheusMetrics {
+    fn publication_headers(&mut self, group: &str) {
+        self.header(
+            "valid_points", "gauge",
+            format_args!("valid publication points {}", group)
+        );
+        self.header(
+            "rejected_points", "gauge",
+            format_args!("rejected publication points {}", group)
+        );
+        self.header(
+            "valid_manifests", "gauge",
+            format_args!("valid manifests {}", group)
+        );
+        self.header(
+            "invalid_manifests", "gauge",
+            format_args!("invalid manifests {}", group)
+        );
+        self.header(
+            "stale_manifests", "gauge",
+            format_args!("stale manifests {}", group)
+        );
+        self.header(
+            "missing_manifests", "gauge",
+            format_args!("missing manifests {}", group)
+        );
+        self.header(
+            "valid_crls", "gauge",
+            format_args!("valid CRLs {}", group)
+        );
+        self.header(
+            "invalid_crls", "gauge",
+            format_args!("invalid CRLs {}", group)
+        );
+        self.header(
+            "stale_crls", "gauge",
+            format_args!("stale CRLs {}", group)
+        );
+        self.header(
+            "stray_crls", "gauge",
+            format_args!("stray CRLs {}", group)
+        );
+        self.header(
+            "valid_ca_certs", "gauge",
+            format_args!("valid CA certificates {}", group)
+        );
+        self.header(
+            "valid_ee_certs", "gauge",
+            format_args!("valid router certificates {}", group)
+        );
+        self.header(
+            "invalid_certs", "gauge",
+            format_args!("invalid certificate files {}", group)
```

Comment:
consistency wrt the `valid_*_certs` headers :
```suggestion
            format_args!("invalid certificates {}", group)
```

Reply:
This was intentional to make clear that this does only count invalid stand-alone certificates rather than those certificates included in other objects (like ROAs).